packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -591,6 +598,39 @@ export class CfnParser {
     }
   }
 
+  private handleRulesIntrinsic(key: string, object: any): any {
+    // Rules have their own set of intrinsics:
+    // https://docs.aws.amazon.com/servicecatalog/latest/adminguide/reference-template_constraint_rules.html
+    switch (key) {
+      case 'Fn::ValueOf': {
+        // ValueOf is special,
+        // as it takes the name of a Parameter as its first argument
+        const value = this.parseValue(object[key]);
+        const parameterName = value[0];
+        if (parameterName in this.parameters) {
+          // since ValueOf returns the value of a specific attribute,
+          // fail here - this substitution is not allowed
+          throw new Error(`Cannot substitute parameter '${parameterName}' used in Fn::ValueOf expression with attribute '${value[1]}'`);
+        }
+        const param = this.options.finder.findRefTarget(parameterName);
+        if (!param) {
+          throw new Error(`Rule references parameter '${parameterName}' which was not found in the template`);
+        }
+        // create an explicit IResolvable,
+        // as Fn.valueOf() returns a string,
+        // which is incorrect
+        // (Fn::ValueOf can also return an array)
+        return Lazy.anyValue({ produce: () => ({ 'Fn::ValueOf': [param.logicalId, value[1]] }) });
+      }
+      default:
+        // all other intrinsics,
+        // to not hard-code the list of supported ones,
+        // we treat as simple objects,
+        // by returning undefined here
```

Comment:
_(Nit):_ I'm not great with poetry. How about the simpler:
```suggestion
        // Treat all other intrinsics as simple objects
```

Reply:
Alright, I admit this wasn't the best comment I've ever written :). Changed!